incubator/hnc/controllers/object_controller_test.go

```diff
@@ -52,6 +55,59 @@ var _ = Describe("Secret", func() {
 		Eventually(hasSecret(ctx, bazName, "bar-sec")).Should(BeFalse())
 		Eventually(hasSecret(ctx, bazName, "foo-sec")).Should(BeFalse())
 	})
+
+	It("should be marked as modified if not matching the source", func() {
+		// Set tree as qux -> baz -> bar and make sure the propagation of bar-sec is
+		// *fully finished* before modifying the bar-sec in baz namespace
+		setParent(ctx, bazName, barName)
+		setParent(ctx, quxName, bazName)
+		Eventually(hasSecret(ctx, bazName, "bar-sec")).Should(BeTrue())
+		Eventually(hasSecret(ctx, quxName, "bar-sec")).Should(BeTrue())
+
+		Eventually(hasAnnotationModified(ctx, bazName, "bar-sec")).Should(BeFalse())
+		Eventually(hasAnnotationModified(ctx, bazName, "baz-sec")).Should(BeFalse())
+
+		modifySecret(ctx, bazName, "bar-sec")
+
+		// Wait 3 more seconds for the annotation to be updated
```

Comment:
I tried even 4 seconds. It cannot always pass. It could occasionally get a timed out before the hasAnnotationModified set to be true. The default timeout for Eventually() is 2 seconds, so I tried 5s. Is it a concern that it took this long?
Also, is it better to set a custom timeout than use time.sleep()? Since it will hit and proceed before the full timeout.

Reply:
There may be another racing updates between modify and then setParent(). I will look.

Reply:
Ok we can leave it for now, but let's have a look again later. Maybe add a comment that it didn't work at 4s and you needed to increase it, and add a TODO to see why it's so slow.